Add mediawiki import to docs (#1348)
django-wiki/django-wiki: docs/tips/mediawiki.rst

Mediawiki
=========

If you want to import from Mediawiki you can create an XML dump of the pages and then use
a management command for your app to import it.

The following management command code should work fine with the latest django-wiki version and
latest mediawiki version.
It uses the lxml library to parse the mediawiki xml
and the unidecode to convert non-latin characters to ascii (so as to create their slug). Finally it

> uses pandoc to do the actual markdown -> github flavored markdown convert (I have tested it on windows and it works great).

Reviewer: ```suggestion
uses pandoc to do conversion from MediaWiki markup to GitHub Flavored Markdown (which in this case renders fine in django-wiki).
```

        def add_arguments(self, parser):
            parser.add_argument("file", type=str)

        @transaction.atomic()
        def handle(self, *args, **options):
            user = get_user_model().objects.get(username="root")
            current_site = Site.objects.get_current()
            url_root = URLPath.root()

            tree = etree.parse(options["file"])
            pages = tree.xpath('// *[local-name()="page"]')
            for p in pages:
                title = p.xpath('*[local-name()="title"]')[0].text
                print(title)
                revision = p.xpath('*[local-name()="revision"]')[0]
                text = revision.xpath('*[local-name()="text"]')[-1].text
                timestamp = revision.xpath('*[local-name()="timestamp"]')[0].text
                timestamp = datetime.datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%SZ")
                timestamp_with_timezone = pytz.utc.localize(timestamp)

                import_page(
                    current_site,
                    url_root,
                    text,
                    title,
                    timestamp_with_timezone,
                    True,
                    user,
            )

> 

Reviewer: ```suggestion

Usage
-----

Once the management command is provided by your Django application, you can invoke it from the command-line:

.. code-block:: console

   python manage.py import_mediawiki_dump <mediawiki-xml-dump-file>``

```

Mediawiki
=========

If you want to import articles from Mediawiki, you can create an XML dump of the pages and then use
a Django management command to import it.

> The management command is not provided as part of django-wiki, but we'll show you how to build on for your own Django app.

Reviewer: ```suggestion
The management command is not provided as part of django-wiki, but we'll show you how to build one for your own Django app.
```

Mistake in my own text :blush:
